diff --git a/node_modules/react-native-css-interop/dist/metro/index.js b/node_modules/react-native-css-interop/dist/metro/index.js
index c91ba6a..09cb8e0 100644
--- a/node_modules/react-native-css-interop/dist/metro/index.js
+++ b/node_modules/react-native-css-interop/dist/metro/index.js
@@ -177,17 +177,21 @@ async function startCSSProcessor(filePath, platform, isDev, { input, getCSSForPl
                 : getNativeJS((0, css_to_rn_1.cssToReactNativeRuntime)(css, options), debug)));
             debug(`virtualStyles.emit ${platform}`);
             haste.emit("change", {
-                eventsQueue: [
-                    {
-                        filePath,
-                        metadata: {
-                            modifiedTime: Date.now(),
-                            size: 1,
-                            type: "virtual",
-                        },
-                        type: "change",
-                    },
-                ],
+                changes: {
+                    addedFiles: new Map(),
+                    modifiedFiles: new Map([
+                        [
+                            filePath,
+                            {
+                                modifiedTime: Date.now(),
+                                size: 1,
+                                type: "virtual",
+                            },
+                        ],
+                    ]),
+                    removedFiles: new Map(),
+                },
+                rootDir: "",
             });
         }).then((css) => {
             debug(`virtualStyles.initial ${platform}`);
diff --git a/node_modules/react-native-css-interop/src/metro/index.ts b/node_modules/react-native-css-interop/src/metro/index.ts
index fa58c7b..bb72dfc 100644
--- a/node_modules/react-native-css-interop/src/metro/index.ts
+++ b/node_modules/react-native-css-interop/src/metro/index.ts
@@ -342,17 +342,21 @@ async function startCSSProcessor(
 
         debug(`virtualStyles.emit ${platform}`);
         haste.emit("change", {
-          eventsQueue: [
-            {
-              filePath,
-              metadata: {
-                modifiedTime: Date.now(),
-                size: 1, // Can be anything
-                type: "virtual", // Can be anything
-              },
-              type: "change",
-            },
-          ],
+          changes: {
+            addedFiles: new Map(),
+            modifiedFiles: new Map([
+              [
+                filePath,
+                {
+                  modifiedTime: Date.now(),
+                  size: 1, // Can be anything
+                  type: "virtual", // Can be anything
+                },
+              ],
+            ]),
+            removedFiles: new Map(),
+          },
+          rootDir: "",
         });
       }).then((css) => {
         debug(`virtualStyles.initial ${platform}`);
